File: prototype/Assets/microcosmicWar/Scripts/test/mainInput.js

var actionCommandControl:ActionCommandControl;

function setToControl(pActionCommandControl:ActionCommandControl)
{
	actionCommandControl = pActionCommandControl;
}

function GetActionCommandFromInput():UnitActionCommand
{
	var lActionCommand=UnitActionCommand();
	//lActionCommand.MoveLeft=Input.GetButton ("left");
	if(Input.GetButton ("left"))
	{
		lActionCommand.FaceLeft=true;
		lActionCommand.GoForward=true;
	}
	//lActionCommand.MoveRight=Input.GetButton ("right");
	if(Input.GetButton ("right"))
	{
		lActionCommand.FaceRight=true;
		lActionCommand.GoForward=true;
	}
	if(Input.GetButton ("down"))
		lActionCommand.FaceDown=true;
	if(Input.GetButton ("up"))
		lActionCommand.FaceUp=true;
		
	lActionCommand.Fire=Input.GetButton ("fire");
	lActionCommand.Jump=Input.GetButtonDown("jump");
	return lActionCommand;
}

function Update ()
{
	//if( zzCreatorUtility.isHost())
	if(actionCommandControl)
		actionCommandControl.setCommand(GetActionCommandFromInput());
}

File: prototype/Assets/microcosmicWar/Scripts/test/mainInput.cs

using UnityEngine;
using System.Collections;

public class mainInput : MonoBehaviour
{
    [SerializeField]
    CommandControlBase[] _actionCommandControl;
    public CommandControlBase[] actionCommandControl
    {
        get
        {
            return _actionCommandControl;
        }
        set
        {
            _actionCommandControl = value;
        }
    }
    public ObjectOperatives objectOperatives;

    //public void setToControl(CommandControlBase pActionCommandControl)
    //{
    //    _actionCommandControl = pActionCommandControl;
    //}
    readonly CommandControlBase[] nullActionCommandControl = new CommandControlBase[] { };
    public CommandControlBase[] clearCommand()
    {
        foreach (var lControl in _actionCommandControl)
        {
            lControl.setCommand(new UnitActionCommand());
        }
        var lOut = _actionCommandControl;
        _actionCommandControl = nullActionCommandControl;
        return lOut;
    }

    public UnitActionCommand GetActionCommandFromInput()
    {
        UnitActionCommand lActionCommand = new UnitActionCommand();
        //lActionCommand.MoveLeft=Input.GetButton ("left");
        if (Input.GetButton("left"))
        {
            lActionCommand.FaceLeft = true;
            lActionCommand.GoForward = true;
        }
        //lActionCommand.MoveRight=Input.GetButton ("right");
        if (Input.GetButton("right"))
        {
            lActionCommand.FaceRight = true;
            lActionCommand.GoForward = true;
        }
        lActionCommand.FaceDown = Input.GetButton("down");
        lActionCommand.FaceUp = Input.GetButton("up");

        lActionCommand.Fire = Input.GetButton("fire");
        //lActionCommand.Jump=Input.GetButtonDown("jump");
        lActionCommand.Jump = Input.GetButton("jump");
        lActionCommand.Action1 = Input.GetButton("action1");
        lActionCommand.Action2 = Input.GetButton("action2");
        return lActionCommand;
    }

    void Update()
    {
        //if( zzCreatorUtility.isHost())
        //if (_actionCommandControl)
        //    _actionCommandControl.setCommand(GetActionCommandFromInput());
        if (_actionCommandControl.Length > 0)
        {
            var lCommand = GetActionCommandFromInput();
            foreach (var lControl in _actionCommandControl)
            {
                lControl.setCommand(lCommand);
            }
        }

        if (objectOperatives && Input.GetButtonDown("function"))
            objectOperatives.doOperate();
    }
}